File: 3877-minimum-removals-to-achieve-target-xor/3877-minimum-removals-to-achieve-target-xor.py
class Solution:
    def minRemovals(self, nums: List[int], target: int) -> int:
        n = len(nums)
        
        dp = {0: 0}   # xor -> max length
        
        for num in nums:
            new_dp = dp.copy()
            for xor_val, length in dp.items():
                new_xor = xor_val ^ num
                new_dp[new_xor] = max(new_dp.get(new_xor, 0), length + 1)
            dp = new_dp
        
        if target not in dp:
            return -1
        
        return n - dp[target]

File: 3877-minimum-removals-to-achieve-target-xor/3877-minimum-removals-to-achieve-target-xor.cpp
class Solution {
public:
    int minRemovals(vector<int>& nums, int target) {
        int n=nums.size(),MAXX=1e5;
        vector<int> dp(MAXX+1,-1); // dp[i]=max number of elements to take such that XOR = i
        dp[0]=0;
        for(int x:nums){
            vector<int> ndp=dp;
            for(int i=0;i<=MAXX;i++){
                if(dp[i]!=-1){ // [....] = XOR = i
                    ndp[i^x]=max(ndp[i^x],dp[i]+1);
                }
            }
            dp=ndp;
        }
        if(dp[target]==-1) return -1;
        return n-dp[target];
    }
};